src/components/NftRewards/NftRewardsSection.tsx

```diff
@@ -72,11 +72,27 @@ export function NftRewardsSection() {
   const payAmountETH =
     payInCurrency === ETH ? payAmount : fromWad(converter.usdToWei(payAmount))
 
-  const handleTierDeselect = (tierId: number | undefined) => {
+  const handleTierDeselect = (
+    tierId: number | undefined,
+    quantity: number | undefined, // quantity to deselect. Remove all instances of tierId is undefined
+  ) => {
     if (tierId === undefined || !rewardTiers || !payMetadata) return
-    const idxToRemove = payMetadata.tierIdsToMint.indexOf(tierId)
-    const newSelectedTierIds = payMetadata.tierIdsToMint
-    newSelectedTierIds.splice(idxToRemove, 1)
+
+    let count = 0
+    const newSelectedTierIds = [...(payMetadata?.tierIdsToMint ?? [])].filter(
+      id => {
+        // remove all instances
+        if (!quantity) {
+          return id !== tierId
+        }
+        // remove the specified number of instances of tierId
+        if (count < quantity && id === tierId) {
+          count++
+          return false
+        }
+        return true
+      },
+    )
```

Comment:
**issue** filters shouldn't have side effects - a filters job is purely to filter a list and nothing else. Is there a way this can be rewritten in a way that avoids this `count` var?

Reply:
@johnnyd-eth wdyt about this as a slightly less performant but easier to read alt?

```diff
diff --git a/src/components/NftRewards/NftRewardsSection.tsx b/src/components/NftRewards/NftRewardsSection.tsx
index 736abe506..fd75fe756 100644
--- a/src/components/NftRewards/NftRewardsSection.tsx
+++ b/src/components/NftRewards/NftRewardsSection.tsx
@@ -54,10 +54,12 @@ export function NftRewardsSection() {
   const { form: payProjectForm } = useContext(PayProjectFormContext)
   const { projectMetadata } = useContext(ProjectMetadataContext)
   const { fundingCycleMetadata } = useContext(V2V3ProjectContext)
-  const isMobile = useMobile()
 
+  const isMobile = useMobile()
   const { visible: nftPostPayModalVisible, hide: hideNftPostPayModal } =
     useModalFromUrlQuery(NFT_PAYMENT_CONFIRMED_QUERY_PARAM)
+  const converter = useCurrencyConverter()
+
   const {
     payAmount,
     payMetadata,
@@ -67,8 +69,6 @@ export function NftRewardsSection() {
     setPayMetadata,
     validatePayAmount,
   } = payProjectForm ?? {}
-
-  const converter = useCurrencyConverter()
   const payAmountETH =
     payInCurrency === ETH ? payAmount : fromWad(converter.usdToWei(payAmount))
 
@@ -78,19 +78,9 @@ export function NftRewardsSection() {
   ) => {
     if (tierId === undefined || !rewardTiers || !payMetadata) return
 
-    let count = 0
-    const newSelectedTierIds = (payMetadata?.tierIdsToMint ?? []).filter(id => {
-      // remove all instances
-      if (!quantity) {
-        return id !== tierId
-      }
-      // remove the specified number of instances of tierId
-      if (count < quantity && id === tierId) {
-        count++
-        return false
-      }
-      return true
-    })
+    const newSelectedTierIds = (payMetadata?.tierIdsToMint ?? [])
+      .filter(id => id !== tierId)
+      .concat(Array(quantity).fill(tierId))
 
     setPayMetadata?.({
       tierIdsToMint: newSelectedTierIds,

```